lib/express/server.js

```diff
@@ -93,7 +94,18 @@ async function server (opts = {}) {
         }
       });
     });
-    configureServer(app, routeConfiguringFunction, allowCors, basePath);
+
+    configureServer({app, addRoutes: routeConfiguringFunction, allowCors, basePath, plugins});
+
+    // allow plugins to update the app and http server objects
+    for (const plugin of plugins.filter((p) => p.updatesServer)) {
+      log.info(`Allowing plugin ${plugin.name} to modify the Appium server`);
+      plugin.updateServer(app, httpServer).catch(reject);
```

Comment:
should we check first if the plugin has `updateServer` method and this method returns a promise? should this method run asynchronously?

Reply:
Yeah, good idea, for better error messages.

Reply:
Oh, and you're right, we want to run those updates synchronously, so I'll need to refactor this a bit.